datafusion/core/src/execution/context.rs

```diff
@@ -1887,26 +1885,6 @@ mod tests {
         Ok(())
     }
 
-    #[tokio::test]
-    async fn ctx_sql_should_optimize_plan() -> Result<()> {
-        let ctx = SessionContext::new();
-        let plan1 = ctx
-            .create_logical_plan("SELECT * FROM (SELECT 1) AS one WHERE TRUE AND TRUE")?;
-
-        let opt_plan1 = ctx.optimize(&plan1)?;
-
-        let plan2 = ctx
-            .sql("SELECT * FROM (SELECT 1) AS one WHERE TRUE AND TRUE")
-            .await?;
-
-        assert_eq!(
```

Comment:
Maybe we could address the UX by making `to_logical_plan` call optimize?

Reply:
Ah, yes, I like that idea.